### PR TITLE
[feat]: [PIE-3434]: Capturing pipeline execution start and preflight execution

### PR DESCRIPTION
**Pull request by brijeshdhakar**
…execution

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32144)

### CODE REVIEW

File: 800-pipeline-service/src/main/java/io/harness/pms/plan/execution/PipelineExecutor.java [hunk 1]
```diff
@@ -42,11 +47,13 @@
 @AllArgsConstructor(access = AccessLevel.PACKAGE, onConstructor = @__({ @Inject }))
 @Slf4j
 public class PipelineExecutor {
+  private final String PIPELINE_EXECUTION_START_EVENT = "ng_pipeline_execution_start";
```
Comment (naidu-harness) [inline 800-pipeline-service/src/main/java/io/harness/pms/plan/execution/PipelineExecutor.java:50]: ng_start_pipeline_execution?

Comment (brijeshdhakar) [reply 800-pipeline-service/src/main/java/io/harness/pms/plan/execution/PipelineExecutor.java]: Yes, this looks better. Updated te event name.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 2 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32144)* status: Some of the checks are too old. 8 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[800-pipeline-service/src/main/java/io/harness/pms/instrumentaion/PipelineTelemetryHelper.java, line 43 at r2](https://reviewable.io/reviews/harness/harness-core/32144#-N-M90bTCwPkwqPM4hTK:-N-M90bTCwPkwqPM4hTL:bjtkrym) ([raw file](https://github.com/harness/harness-core/blob/448c4e35c3cf49a367d2942113233d835733b901/800-pipeline-service/src/main/java/io/harness/pms/instrumentaion/PipelineTelemetryHelper.java#L43)):*
> ```Java
> 
>   protected void sendTelemetryEventInternal(String eventName, String accountId, HashMap<String, Object> properties) {
>     AccountDTO accountDTO = accountService.getAccount(accountId);
> ```

We can cache accountId to Name mapping here I think.

**Review by vaibhav-si — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32144)* status: Some of the checks are too old. 8 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[800-pipeline-service/src/main/java/io/harness/pms/instrumentaion/PipelineTelemetryHelper.java, line 43 at r2](https://reviewable.io/reviews/harness/harness-core/32144#-N-M90bTCwPkwqPM4hTK:-N-MBuBZ7zfAb7mZn1ZE:b-vsd3g9) ([raw file](https://github.com/harness/harness-core/blob/448c4e35c3cf49a367d2942113233d835733b901/800-pipeline-service/src/main/java/io/harness/pms/instrumentaion/PipelineTelemetryHelper.java#L43)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

We can cache accountId to Name mapping here I think.
</blockquote></details>

This cache we can use for other telemetry events as well like API Telemetry

**Comment by brijeshdhakar** [comment]
*[800-pipeline-service/src/main/java/io/harness/pms/instrumentaion/PipelineTelemetryHelper.java, line 43 at r2](https://reviewable.io/reviews/harness/harness-core/32144#-N-M90bTCwPkwqPM4hTK:-N-MC_WyEEECYQOS6oqP:b-1wnlp7) ([raw file](https://github.com/harness/harness-core/blob/448c4e35c3cf49a367d2942113233d835733b901/800-pipeline-service/src/main/java/io/harness/pms/instrumentaion/PipelineTelemetryHelper.java#L43)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

This cache we can use for other telemetry events as well like API Telemetry
</blockquote></details>

yes Vaibhav. It makes sense. We are using this at many places. I am creating a ticket for this and we can discuss and add the cache.

**Review by archit-harness — APPROVED** [review]
> Reviewed 7 of 8 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32144)* status: Some of the checks are too old. 6 file(s) to review. 2. unresolved discussions.

**Review by vaibhav-si — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32144)* status: 1. unresolved discussions.

**Review by vaibhav-si — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32144#-:-N-MfHRrCxnvwLbUwkpa:bnfp4nl)* status: 1. unresolved discussions.

**Pull request merged**